lib/logstasher/railtie.rb

```diff
@@ -41,6 +41,20 @@ class Railtie < Rails::Railtie
     end
   end
 
+  def default_source
+    case RUBY_PLATFORM
+    when /darwin/
+      # NOTE: MacOS Sierra and later are setting `.local`
+      # hostnames that even as real hostnames without the `.local` part,
+      # are still unresolvable. One reliable way to get an IP is to
+      # get all available IP address lists and use the first one.
+      # This will always be `127.0.0.1`.
+      Socket.ip_address_list.first
```

Comment:
This returns an instance of `Addrinfo`, not a string!
Use `Socket.ip_address_list.first.ip_address` instead!